src/components/utrecht/TableCell.test.tsx

```diff
@@ -0,0 +1,137 @@
+   * because React doesn't allow standalone rendering of <td>.
+   *
+   * Since simply using `:only-child` like the other tests doesn't work anymore,
+   * the following tests heavily rely on `useRef()`. (That's also why the test for ForwardRef is
+   * one of the first tests.)
+   */
+  it("renders a cell role element", () => {
+    render(
+      <Table>
+        <TableBody>
+          <TableRow>
+            <TableCell />
+          </TableRow>
+        </TableBody>
+      </Table>
+    );
+
+    const cell = screen.getByRole("cell");
+
+    expect(cell).toBeInTheDocument();
+    expect(cell).toBeVisible();
+  });
+
+  it("supports ForwardRef in React", () => {
+    const ref = createRef<HTMLTableCellElement>();
+
+    render(
+      <Table>
+        <TableBody>
+          <TableRow>
+            <TableCell ref={ref} />
+          </TableRow>
+        </TableBody>
+      </Table>
+    );
+
+    const row = screen.getByRole("row");
+    const cell = row.querySelector(":only-child");
+
+    expect(ref.current).toBe(cell);
+  });
+
+  it("renders a td HTML element", () => {
+    const { container } = render(
+      <Table>
+        <TableBody>
+          <TableRow>
+            <TableCell />
+          </TableRow>
+        </TableBody>
+      </Table>
+    );
+    const cell = container.querySelector("td:only-child");
+
+    expect(cell).toBeInTheDocument();
+  });
+
+  it("renders a design system BEM class name", () => {
+    const ref = createRef<HTMLTableCellElement>();
+    render(
+      <Table>
+        <TableBody>
+          <TableRow>
+            <TableCell ref={ref} />
+          </TableRow>
+        </TableBody>
+      </Table>
+    );
+    const cell = ref.current;
+
+    expect(cell).toHaveClass("utrecht-table__cell");
+  });
+
+  it("renders rich text content", () => {
+    const ref = createRef<HTMLTableCellElement>();
+    render(
+      <Table>
+        <TableBody>
+          <TableRow>
+            <TableCell ref={ref}>
+              <data value="1234">1,234</data>
+            </TableCell>
+          </TableRow>
+        </TableBody>
+      </Table>
+    );
+
+    const cell = ref.current;
+
+    const richText = cell?.querySelector("data");
+
+    expect(richText).toBeInTheDocument();
+  });
+
+  it("can be hidden", () => {
+    const ref = createRef<HTMLTableCellElement>();
+    render(
+      <Table>
+        <TableBody>
+          <TableRow>
+            <TableCell ref={ref} hidden />
+          </TableRow>
+        </TableBody>
+      </Table>
+    );
+
+    expect(ref.current).not.toBeVisible();
+  });
+
+  it("can have a custom class name", () => {
+    const ref = createRef<HTMLTableCellElement>();
+    render(
+      <Table>
+        <TableBody>
+          <TableRow>
+            <TableCell ref={ref} className="negative">
+              -42
+            </TableCell>
+          </TableRow>
+        </TableBody>
+      </Table>
+    );
+
+    expect(ref.current).toHaveClass("negative");
```

Comment:
As with the visible comment, I do also like this ref check but should we be consistent and use either ref or queries to select the rendered component everywhere?

Reply:
Also refactored this to `data-testid` instead of `ref`.